swarm-derive/src/lib.rs

```diff
@@ -181,35 +190,44 @@ fn build_struct(ast: &DeriveInput, data_struct: &DataStruct) -> TokenStream {
             })
     };
 
-    // Build the list of statements to put in the body of `inject_connection_established()`.
-    let inject_connection_established_stmts = {
-        data_struct.fields.iter().enumerate().map(move |(field_n, field)| {
-            match field.ident {
-                Some(ref i) => quote!{ self.#i.inject_connection_established(peer_id, connection_id, endpoint, errors, other_established); },
-                None => quote!{ self.#field_n.inject_connection_established(peer_id, connection_id, endpoint, errors, other_established); },
-            }
-        })
+    // Build the list of statements to put in the body of `on_swarm_event()`
+    // for the FromSwarm::ConnectionEstablished variant.
+    let on_connection_established_stmts = {
+        data_struct
+            .fields
+            .iter()
+            .enumerate()
+            .map(|(field_n, field)| match field.ident {
+                Some(ref i) => quote! { self.#i.on_swarm_event(
+                    #from_swarm::ConnectionEstablished(#connection_established { peer_id, connection_id, endpoint, failed_addresses, other_established}));},
+                None => quote! { self.#field_n.on_swarm_event(
+                    #from_swarm::ConnectionEstablished(#connection_established { peer_id, connection_id, endpoint, failed_addresses, other_established}));},
+            })
     };
 
-    // Build the list of statements to put in the body of `inject_address_change()`.
-    let inject_address_change_stmts = {
-        data_struct.fields.iter().enumerate().map(move |(field_n, field)| {
-            match field.ident {
-                Some(ref i) => quote!{ self.#i.inject_address_change(peer_id, connection_id, old, new); },
-                None => quote!{ self.#field_n.inject_address_change(peer_id, connection_id, old, new); },
-            }
-        })
+    // Build the list of statements to put in the body of `on_swarm_event()`
+    // for the FromSwarm::AddressChange variant.
+    let on_address_change_stmts = {
+        data_struct
+            .fields
+            .iter()
+            .enumerate()
+            .map(|(field_n, field)| match field.ident {
+                Some(ref i) => quote! { self.#i.on_swarm_event(#from_swarm::AddressChange(#address_change { peer_id, connection_id, old, new }));},
+                None => quote! { self.#field_n.on_swarm_event(#from_swarm::AddressChange(#address_change { peer_id, connection_id, old, new }));},
+            })
     };
 
-    // Build the list of statements to put in the body of `inject_connection_closed()`.
-    let inject_connection_closed_stmts = {
+    // Build the list of statements to put in the body of `on_swarm_event()`
+    // for the FromSwarm::ConnectionClosed variant.
```

Comment:
```suggestion
    // for the `FromSwarm::ConnectionClosed` variant.
```

Nit pick, for the sake of consistency.

Reply:
thanks Max! addressed this and the remaining cases.